### DISCUSSION TITLE
比较推荐的学习路线 | HackWay技术学习路线

### DISCUSSION THREAD

**Discussion by giscus** [lidongyx/hackwaydoc#56] [Announcements]
# 比较推荐的学习路线 | HackWay技术学习路线

计算机专业所涉及的课程以及方向非常多，HackWay会尽可能详实地整理6大CS名校的所有关键课程，但这些课程对于个人学习者来说未免过于庞杂；而且相同的课程不同学校都有授课，到底该如何取舍？而且不少课程资源配套以及开放的程度不一，哪一门课提供的更加丰富...等等，在这里综合多方面的考量为大家提供一个值得参考的学习路线，既做到循序渐进，又不缺失核心的经典课程，而且课程的资源也比较容易获取。

https://hackway.org/docs/cs/map

**Comment by lengzer** [comment]
《大一上学期》中“学习地址：MIT 计算机教育中缺失的一课”这里的URL点不开，是本地地址哈

**Reply by lidongyx** [reply]
网站进行了一波升级改版，内容会不断完善更新，谢谢之前指出错误的地方

**Comment by BriefAnswer** [comment]
关于课程编号的介绍有问题，在网上找到了比较合理的介绍：学校是根据课程的深度给课程排号的。如计算机系的课号是CS0到CS599，其中：0\~99号是服务性课程，适合非技术性专业的学生选择;100\~199号是本科生基础课程;200\~299号是高级本科生课程/初级研究生课程;300~399号是高级研究生课程;后边的课号代表实验和讨论班。（参考文章 [斯坦福大学本科计算机专业学习哪些课程？](https://www.topsedu.com/about/news/20546.html)）

**Comment by GooogleLovers** [comment]
你好HackWay樓主,我前段時間參考了下你的路線圖,真的收益頗深,然後我總結了一個適合所有人的計算機科學路線,我今天發現你也出了個參考路線,我感覺我的和您的比較類似呀哈哈😂,這是計算機科學路線參考自HackWay.org! 豆瓣連結🔗https://www.douban.com/doulist/158374585/

**Reply by GooogleLovers** [reply]
我的建議是時間緊的可以不計算機科學概論和Unix and Linux,然後可以不學離散數學直接算法,在看過深入理解計算機系統後可以不看計算機組成與設計,其實最後面兩個方向分布式和編譯原理真的太難了,可以工作之後補一下,所以就不放進去了,大二全的路線我還有另一個 https://www.douban.com/people/251480614/doulists/all 路線,另外我的豆瓣主頁還有計算機科學的數學路線和C/C++路線

**Comment by yexiaofeng147** [comment]
没有加上技术交流群、请你通过一下
